beacon-chain/rpc/prysm/v1alpha1/validator/proposer.go

```diff
@@ -177,6 +219,135 @@ func (vs *Server) ProposeBlock(ctx context.Context, rBlk *ethpb.SignedBeaconBloc
 	}, nil
 }
 
+// GetBlockAltair is called by a proposer during its assigned slot to request a block to sign
+// by passing in the slot and the signed randao reveal of the slot. This is used by a validator
+// after the altair fork epoch has been encountered.
+func (vs *Server) GetBlockAltair(ctx context.Context, req *ethpb.BlockRequest) (*ethpb.BeaconBlockAltair, error) {
+	ctx, span := trace.StartSpan(ctx, "ProposerServer.GetBlockAltair")
+	defer span.End()
+	span.AddAttributes(trace.Int64Attribute("slot", int64(req.Slot)))
+
+	if helpers.SlotToEpoch(req.Slot) < params.BeaconConfig().AltairForkEpoch {
```

Comment:
To be honest, this seems like something to be handled by the caller, not the beacon server.  If we add this to this method, would that mean all altair methods need to have this check ?

Reply:
I think the error returned by this check is very meaningful and helpful.

> If we add this to this method, would that mean all altair methods need to have this check ?

Yes, unless we handle Altair methods the same way official APIs does. Their `v2` endpoints return both versions e.g. https://github.com/prysmaticlabs/prysm/blob/265b5feabfe5ca00367b2b58efba7900e25e28e5/beacon-chain/rpc/eth/debug/debug.go#L57

Reply:
We would need to use `one of` over here then. Also the schema for this RPC method defers from our other ones. Maybe we should be using `oneOf` over here ? That way a merge block can also be added in without having to re-implement a new method altogether for block proposals.

Reply:
Returning errors will be much less time consuming. Should we take this route now and revisit the implementation later?